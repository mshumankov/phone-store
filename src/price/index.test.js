import React from 'react';
import Price from './index';
import renderer from 'react-test-renderer';

describe('Price component', () => {
    test('should render correctly', () => {
        const component = renderer.create(
            <Price />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('should render correctly if price is 100$', () => {
        const component = renderer.create(
            <Price priceAll={100} />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('should render correctly if price is 100$ and priceMonth = true', () => {
        const component = renderer.create(
            <Price priceAll={100} priceMonth={true} />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('should render correctly if update price with 100$ and priceMonth = true', () => {
        let component;
        renderer.act(() => {
            component = renderer.create(<Price />)
        });

        expect(component.toJSON()).toMatchSnapshot();

        renderer.act(() => {
            component.update(<Price priceAll={100} />)
        });

        expect(component.toJSON()).toMatchSnapshot();

        renderer.act(() => {
            component.update(<Price priceAll={100} priceMonth={true} />)
        });

        expect(component.toJSON()).toMatchSnapshot();
    })


})