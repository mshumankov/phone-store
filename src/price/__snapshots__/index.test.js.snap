// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component should render correctly 1`] = `null`;

exports[`Price component should render correctly if price is 100$ 1`] = `
<p
  className="price-number"
>
  <span
    className="currency"
  >
    $ 
  </span>
  99,
  <span
    className="coins"
  >
    99 
  </span>
</p>
`;

exports[`Price component should render correctly if price is 100$ and priceMonth = true 1`] = `
<p
  className="price-number"
>
  <span
    className="currency"
  >
    $ 
  </span>
  9,
  <span
    className="coins"
  >
    99 
  </span>
</p>
`;

exports[`Price component should render correctly if update price with 100$ and priceMonth = true 1`] = `null`;

exports[`Price component should render correctly if update price with 100$ and priceMonth = true 2`] = `
<p
  className="price-number"
>
  <span
    className="currency"
  >
    $ 
  </span>
  99,
  <span
    className="coins"
  >
    99 
  </span>
</p>
`;

exports[`Price component should render correctly if update price with 100$ and priceMonth = true 3`] = `
<p
  className="price-number"
>
  <span
    className="currency"
  >
    $ 
  </span>
  9,
  <span
    className="coins"
  >
    99 
  </span>
</p>
`;
